source/scripts/storage/storage.js

```diff
@@ -1,23 +1,58 @@
-//storage {
-//   categories: {cat1:[names],cat2:[names]}
-//   savedLists: {list1:[names],list2:[names]}
-//   recipeName1: json object
-//   recipeName2: json object
+// localStorage = {
+//   categories: { // Used to populate explore page (only ids to reduce storage size)
+//       "breakfast": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "lunch": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   savedRecipes: { // Saved recipe page categories (only ids to reduce storage size)
+//       "favorites": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "list2": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   recipeData: { // Complete storage of available recipes we can query without fetching
+//       "111": "full recipe json object", // these are api results so we never have to modify them
+//       "123": "full recipe json object",
+//       "444": "full recipe json object",
+//       "555": "full recipe json object",
+//       "715": "full recipe json object",
+//       "892": "full recipe json object"
+//   }
 // }
 
+
 // store recipe to local storage
-// add recipe name to category
+// add recipeid to category
 // input: category of api fetch, array of recipe json object
 export function storeRecipeData(category, recipeArray) {
```

Comment:
instead of an array of recipe json objects, maybe first write a function that takes a single recipe json object and adds that data to localStorage. then you can define this function that handles a whole array just by calling the other one on each element

Reply:
I think having a function that stores a single recipe at a time and calling that function for each element in array requires a lot more extra getItem fetch from the database. We need to fetch for 'recipaData', 'categories' in local storage for n times. But if we keep it as it is, we only need to call them 1 time which should be faster when we have a lot of data. If in any case we want to save only 1 recipe we can also just use this same function: storeRecipeData(category,[recipe data])